firstvoices/backend/tests/test_apis/base_api_test.py

```diff
@@ -491,7 +493,9 @@ def test_update_403(self):
         instance = self.create_minimal_instance(site=site, visibility=Visibility.PUBLIC)
 
         response = self.client.put(
-            self.get_detail_endpoint(key=instance.id, site_slug=site.slug),
+            self.get_detail_endpoint(key=instance.slug, site_slug=site.slug)
+            if instance.__class__ == SitePage
+            else self.get_detail_endpoint(key=instance.id, site_slug=site.slug),
```

Comment:
Let's not put a class-specific condition in the base test class. Suggested other way to do this:

self.get_detail_endpoint(key=self.get_lookup_key(instance), ....

Add a default get_lookup_key method that returns instance.id, and override that in the site page tests.

Reply:
Great suggestion, thanks!